independent-projects/arc/runtime/src/test/java/io/quarkus/arc/impl/ArcContainerTest.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.arc.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.List;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.arc.Arc;
+
+public class ArcContainerTest {
+
+    @Test
+    public void testContainer() {
+        InMemoryLogHandler handler = new InMemoryLogHandler();
+        Logger rootLogger = Logger.getLogger("");
+        rootLogger.addHandler(handler);
+        try {
+            System.clearProperty("quarkus.arc.log-no-container");
+            // By default, no action is performed
+            assertNull(Arc.container());
+            assertTrue(handler.records.isEmpty());
+
+            IllegalStateException ise = assertThrows(IllegalStateException.class, () -> Arc.requireContainer());
+            assertTrue(ise.getMessage().contains("ArC container not initialized"));
+
+            // "log" -> log a warning
```

Comment:
This comment is obsolete :-) We no longer check the value of the system property.

Reply:
Ah, yes. It should be `quarkus.arc.log-no-container=true -> log a warning`